var/spack/repos/builtin/packages/trilinos/package.py

```diff
@@ -87,14 +87,22 @@ class Trilinos(CMakePackage):
             description='Enables the build of shared libraries')
     variant('debug',        default=False,
             description='Builds a debug version of the libraries')
-    variant('boost',        default=True, description='Compile with Boost')
-    variant('tpetra',       default=True, description='Compile with Tpetra')
-    variant('exodus', default=False, description='Compile with Exodus from SEACAS')
+    variant('boost',        default=True,  description='Compile with Boost')
+    variant('tpetra',       default=True,  description='Compile with Tpetra')
+    variant('exodus',       default=False, description='Compile with Exodus from SEACAS')
+
+    variant('dtk',          default=False, description='Enable DataTransferKit')
+    resource(name='dtk',
+             git='https://github.com/ornl-cees/DataTransferKit',
```

Comment:
@aprokop the disadvantage of encoding version into mulitvalued variants is that you don't have good control over this from outside. If there is a package which needs `trilinos+dtk`, then it would have to say which version of `dtk` it needs. things like `dtk@1.0:` don't exist there. So i am not sure it's the right direction.